app/code/core/Mage/Eav/Model/Entity/Attribute/Abstract.php

```diff
@@ -133,18 +133,23 @@ protected function _construct()
     public function loadByCode($entityType, $code)
     {
         Varien_Profiler::start('_LOAD_ATTRIBUTE_BY_CODE__');
-        if (is_numeric($entityType)) {
-            $entityTypeId = $entityType;
-        } elseif (is_string($entityType)) {
-            $entityType = Mage::getModel('eav/entity_type')->loadByCode($entityType);
-        }
-        if ($entityType instanceof Mage_Eav_Model_Entity_Type) {
-            $entityTypeId = $entityType->getId();
-        }
-        if (empty($entityTypeId)) {
-            throw Mage::exception('Mage_Eav', Mage::helper('eav')->__('Invalid entity supplied.'));
+        $model = Mage::getSingleton('eav/config')->getAttribute($entityType, $code);
+        if ($model) {
+            $this->setData($model->getData());
```

Comment:
I think a "load" method should still do a clean load rather than copying the cached value to a fresh model.
```suggestion
            $this->load($model->getId());
```

Reply:
I thought same but the problem is the load method is called in same places and the only way an attribute changes is via API/Admin and cache is cleared after that. I'd rather keep it like this, remember the admin area doesn't use cache (only local per-process).

Reply:
Also: even if attribute changes during a request, the _afterSave() methods will call eav/config->clear which resets the cache.

Reply:
I think this would be broken, right?
```php
$attribute = ...->loadByCode('x');
$attribute->setData('foo', 'bar')->save();
$attribute->loadByCode('x');
assertEquals($attribute->getData('foo'), 'bar');
```

Reply:
It shouldn't be, loadByCode calls resource->loadByCode which does a setData($row) which in turn does a $this->_data=$row so I think that assertion should always be false since "re-loading" the model should clear all state.

Reply:
No, save() calls _afterSave() which calls eav/config->clear() 
I did not add that _afterSave call btw, it was present already.

Example stack when saving an attribute:
![image](https://user-images.githubusercontent.com/11006964/216471523-5c042206-2cbc-4737-a2bd-0ceb13fb4ba6.png)

Reply:
It's also important that the cache gets cleared *entirely* otherwise the changes wont be reflected and that might cause issues, which is why I added a `Mage::app()->cleanCache([self::ENTITIES_CACHE_ID]);` to the clear method to flush the entire tag, clearing the cache for all store variants.

Reply:
Ahh ok, thanks. Nice work!

Reply:
Happy to explain it, this PR has substantial internal changes. It has some potential to cause problems which is also why I think it should only go to 20.x and not to 19.x.